config/config.go

```diff
@@ -445,9 +448,24 @@ func initConsensusProtocols() {
 	// v17 can be upgraded to v19.
 	v17.ApprovedUpgrades[protocol.ConsensusV19] = true
 
+	// v20 points to adding the precision to the assets.
+	v20 := v19
+	v20.ApprovedUpgrades = map[protocol.ConsensusVersion]bool{}
+	v20.MaxAssetDecimals = 19
+	// we want to adjust the upgrade time to be roughly one week.
+	// one week, in term of rounds would be:
+	// 140651 = (7 * 24 * 60 * 60 / 4.3)
+	// for the sake of future manual calculations, we'll round that down
+	// a bit :
+	v20.UpgradeWaitRounds = 140000
+	Consensus[protocol.ConsensusV20] = v20
+
+	// v19 can be upgraded to v20.
+	v19.ApprovedUpgrades[protocol.ConsensusV20] = true
```

Comment:
Let's not do this until we have a final version string for `ConsensusV20`.

Reply:
These parts of the PR originate from https://github.com/algorand/go-algorand/pull/576, I will rebase onto that before this is merged